app/src/main/java/com/example/inventory/ui/item/ItemEntryScreen.kt

```diff
@@ -0,0 +1,187 @@
+    override val titleRes = R.string.item_entry_title
+}
+
+@Composable
+fun ItemEntryScreen(
+    navigateBack: () -> Unit,
+    onNavigateUp: () -> Unit,
+    modifier: Modifier = Modifier,
+    canNavigateBack: Boolean = true,
+    viewModel: ItemEntryViewModel = viewModel(factory = AppViewModelProvider.Factory)
+) {
+    val coroutineScope = rememberCoroutineScope()
+    Scaffold(
+        topBar = {
+            InventoryTopAppBar(
+                title = stringResource(ItemEntryDestination.titleRes),
+                canNavigateBack = canNavigateBack,
+                navigateUp = onNavigateUp
+            )
+        }, modifier = modifier
+    ) { innerPadding ->
+        ItemEntryBody(
+            itemUiState = viewModel.itemUiState,
+            onItemValueChange = viewModel::updateUiState,
+            onSaveClick = {
+                // Note: If the user rotates the screen very fast, the operation may get cancelled
+                // and the item may not be saved in the Database. This is because when config
+                // change occurs, the Activity will be recreated and the rememberCoroutineScope will
+                // be cancelled - since the scope is bound to composition.
+                coroutineScope.launch {
+                    viewModel.saveItem()
+                    navigateBack()
+                }
+            },
+            modifier = Modifier.padding(innerPadding)
+        )
+    }
+}
+
+@Composable
+fun ItemEntryBody(
+    itemUiState: ItemUiState,
+    onItemValueChange: (ItemDetails) -> Unit,
+    onSaveClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(dimensionResource(id = R.dimen.padding_medium)),
+        verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.padding_large))
+    ) {
+        ItemInputForm(
+            itemDetails = itemUiState.itemDetails,
+            onValueChange = onItemValueChange,
+            modifier = modifier.fillMaxWidth()
+        )
+        Button(
+            onClick = onSaveClick,
+            enabled = itemUiState.isEntryValid,
+            shape = MaterialTheme.shapes.small,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Text(text = stringResource(R.string.save_action))
+        }
+    }
+}
+
+@Composable
+fun ItemInputForm(
+    itemDetails: ItemDetails,
+    modifier: Modifier = Modifier,
+    onValueChange: (ItemDetails) -> Unit = {},
+    enabled: Boolean = true
+) {
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.padding_medium))
+    ) {
+        OutlinedTextField(
+            value = itemDetails.name,
+            onValueChange = { onValueChange(itemDetails.copy(name = it)) },
+            label = { Text(stringResource(R.string.item_name_req)) },
+            shape = MaterialTheme.shapes.extraSmall,
```

Comment:
if you dont explicitly set this as extraSmall does it not conform to what is in your extra small in your Shapes file?